airflow/www/views.py

```diff
@@ -4408,7 +4408,10 @@ def prefill_form(self, form, pk):
             if is_sensitive and field_name in extra_dictionary:
                 extra_dictionary[field_name] = SENSITIVE_FIELD_PLACEHOLDER
         # form.data is a property that builds the dictionary from fields so we have to modify the fields
-        form.extra.data = json.dumps(extra_dictionary)
+        if extra_dictionary.keys():
+            form.extra.data = json.dumps(extra_dictionary)
+        else:
+            form.extra.data = None
```

Comment:
What happens if we don’t set `form.extra.data` here at all?

Reply:
I didn't check, but my understanding is if there was a non dictionary value in the data field it would remain as that (since we don't unset it on error) and we don't want that.